internal/sync/syncer.go

```diff
@@ -393,3 +409,51 @@ func updateCommitStatus(
 	}
 	return nil
 }
+
+// GetStateCache is the exported method for getting the cache  in other packages
+func (s *Syncer) GetStateCache() LiveStateCache {
+	return s.stateCache
+}
+
+// GetLiveManagedObjects retrieves live managed objects from the provided cache for the given GitSync configuration.
+// It returns a map of Kubernetes resource keys to unstructured objects and an error if any.
+func GetLiveManagedObjects(cache LiveStateCache, gitSync *v1alpha1.GitSync) (map[kube.ResourceKey]*unstructured.Unstructured, error) {
+	var unstructuredObj []*unstructured.Unstructured
+	objs, err := cache.GetManagedLiveObjs(gitSync, unstructuredObj)
+	if err != nil {
+		return nil, err
+	}
+	return objs, nil
+}
+
+// CascadeDeletion performs cascading deletion of live managed objects using GitSync.
+// It utilizes the provided context, Kubernetes client, cache, and GitSync configuration.
+// Returns an error if the deletion process encounters any issues.
+func CascadeDeletion(ctx context.Context, k8sClient client.Client, cache LiveStateCache, gitSync *v1alpha1.GitSync) error {
+	numaLogger := logger.FromContext(ctx)
+	objects, err := GetLiveManagedObjects(cache, gitSync)
+	if err != nil {
+		numaLogger.Debug("Live managed objects not found")
+		return err
+	}
+	err = kubernetes.DeleteManagedObjectsGitSync(ctx, k8sClient, objects)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func ProcessGitSyncDeletion(ctx context.Context, gitSync *v1alpha1.GitSync, s *Syncer) error {
+	config, err := controllerConfig.GetConfigManagerInstance().GetConfig()
+	if err != nil {
+		return err
+	}
+	// if cascade deletion is enabled in the config ,Delete the linked resources to the GitSync
+	if config.CascadeDeletion {
+		err := CascadeDeletion(ctx, s.client, s.stateCache, gitSync)
```

Comment:
nit: `return CascadeDeletion(ctx, s.client, s.stateCache, gitSync)` is good enough here.